### PR TITLE
[CI] Coverity snapshots integration

### PR DESCRIPTION
**Pull request by vialogi**
We'd like to make CI able to post Coverity reports to https://coverity.mellanox.com/
[HPCINFRA-1430](https://jirasw.nvidia.com/browse/HPCINFRA-1430)

### CODE REVIEW

File: contrib/jenkins_tests/cov.sh
Comment (dpressle) [inline contrib/jenkins_tests/cov.sh:1]: Do you use shell check? it seems like you are not. for example, variables should be quoted

Comment (dpressle) [inline contrib/jenkins_tests/cov.sh:1]: We should use bash and not posix (as a team guide line that we want to follow)
if [[ $do_coverity_snapshot = true ]];

File: contrib/jenkins_tests/cov.sh [hunk 1]
```diff
@@ -46,6 +46,15 @@ eval "cov-analyze --config ${cov_dir}/coverity_config.xml \
...
 
+if [ "${do_coverity_snapshot}" == "true" ]; then
+    cov-commit-defects --ssl --on-new-cert trust \
+        --url https://coverity.mellanox.com:8443 \
+        --user ${XLIO_COV_USER} --password ${XLIO_COV_PASSWORD} \
+        --dir "${cov_build}" \
+        --stream libxlio-main \
+        --strip-path ${WORKSPACE}
+fi
+
```
Comment (dpressle) [inline contrib/jenkins_tests/cov.sh:57]: what about failing the job if it didnt work ? like in line 47

**Review by dpressle — CHANGES_REQUESTED** [review]
I also would like to see link to a test run in the comments

**Comment by vialogi** [comment]
Sample test run:
http://hpc-master.lab.mtl.com:8080/job/LibXLIO-opensource/457/

File: contrib/jenkins_tests/cov.sh [hunk 2]
```diff
@@ -46,6 +46,16 @@ eval "cov-analyze --config ${cov_dir}/coverity_config.xml \
...
 
+if [[ "${do_coverity_snapshot}" == true ]]; then
+    cov-commit-defects --ssl --on-new-cert trust \
+        --url https://coverity.mellanox.com:8443 \
+        --user "${XLIO_COV_USER}" --password "${XLIO_COV_PASSWORD}" \
+        --dir "${cov_build}" \
+        --stream libxlio-main \
+        --strip-path "${WORKSPACE}"
+fi
+rc=$(($rc+$?))
```
Comment (dpressle) [inline contrib/jenkins_tests/cov.sh:57]: It can be meaningless here (when we dont run cov commit for example), should be in the if/fi block

**Review by dpressle — CHANGES_REQUESTED** [review]

File: contrib/jenkins_tests/cov.sh (hunk 2 above)
Comment (dpressle) [reply contrib/jenkins_tests/cov.sh:57]: i dont see the fix ?

**Review by dpressle — CHANGES_REQUESTED** [review]

File: contrib/jenkins_tests/cov.sh [hunk 3]
```diff
@@ -46,6 +46,16 @@ eval "cov-analyze --config ${cov_dir}/coverity_config.xml \
...
 rc=$(($rc+$?))
 
+if [[ "${do_coverity_snapshot}" == true ]]; then
+    cov-commit-defects --ssl --on-new-cert trust \
+        --url https://coverity.mellanox.com:8443 \
+        --user "${XLIO_COV_USER}" --password "${XLIO_COV_PASSWORD}" \
+        --dir "${cov_build}" \
+        --stream libxlio-main \
+        --strip-path "${WORKSPACE}"
+rc=$(($rc+$?))
```
Comment (dpressle) [inline contrib/jenkins_tests/cov.sh:56]: Fix Indentation

**Review by dpressle — CHANGES_REQUESTED** [review]

**Pull request merged**